src/foundation.cs

```diff
@@ -8752,7 +8753,7 @@ partial interface NSItemProvider : NSCopying {
 
 		[Mac (10,12)][Async]
 		[Export ("registerCloudKitShareWithPreparationHandler:")]
-		void RegisterCloudKitShare ([BlockCallback] Action<CloudKitRegistrationPreparationHandler> preparationHandler);
+		void RegisterCloudKitShare (CloudKitRegistrationPreparationAction preparationHandler);
```

Comment:
So I did as it asked:

Declare a custom delegate instead of using System.Action / System.Func and add the attribute on the corresponding parameter.

and PR'ed it to start a discussion. If we want to extend generator.cs to remove this restriction:

https://github.com/xamarin/xamarin-macios/blob/master/src/generator.cs#L1282

Else we can just shut up the warning w\ an suppress and fix in XAMCORE_4_0